airbyte-integrations/connectors/source-notion/source_notion/streams.py

```diff
@@ -298,3 +298,65 @@ def should_retry(self, response: requests.Response) -> bool:
             else:
                 return super().should_retry(response)
         return super().should_retry(response)
+
+
+class Comments(HttpSubStream, IncrementalNotionStream):
+    """
+    Comments Object Docs: https://developers.notion.com/reference/comment-object
+    Comments Endpoint Docs: https://developers.notion.com/reference/retrieve-a-comment
+    """
+
+    http_method = "GET"
+    # We can use the "last edited time" of the parent Page as the cursor field,
+    # since we cannot guarantee the order of comments between pages.
+    cursor_field = "page_last_edited_time"
+
+    def path(self, **kwargs) -> str:
+        return f"comments"
+
+    def request_params(self, next_page_token: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        block_id = stream_slice.get("block_id", "")
+        params = {"block_id": block_id, "page_size": self.page_size}
+
+        if next_page_token:
+            params["start_cursor"] = next_page_token["next_cursor"]
+
+        return params
+    
+    def parse_response(self, response: requests.Response, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, **kwargs) -> Iterable[Mapping]:
+        
+        # Get the parent's "last edited time" to compare against state
+        page_last_edited_time = stream_slice.get("page_last_edited_time", "")
+        records = response.json().get("results", [])
+
+        for record in records:
+            record["page_last_edited_time"] = page_last_edited_time
+            state_last_edited_time = stream_state.get(self.cursor_field, "")
+
+            if isinstance(state_last_edited_time, StateValueWrapper):
+                state_last_edited_time = state_last_edited_time.value
+
+            if not stream_state or page_last_edited_time >= state_last_edited_time:
+                yield from transform_properties(record)
+
+    def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
+
+        yield from IncrementalNotionStream.read_records(self, **kwargs)
+    
+    def stream_slices(
+        self, sync_mode: SyncMode, cursor_field: Optional[List[str]] = None, stream_state: Optional[Mapping[str, Any]] = None
+    ) -> Iterable[Optional[Mapping[str, Any]]]:
+                
+        # Gather parent stream records in full
+        parent_records = self.parent.read_records(
+            sync_mode=SyncMode.full_refresh, cursor_field=self.parent.cursor_field, stream_state=stream_state
```

Comment:
if we're syncing full refresh is there a reason to pass state?

Reply:
Nope, that's my bad! I'll remove the argument